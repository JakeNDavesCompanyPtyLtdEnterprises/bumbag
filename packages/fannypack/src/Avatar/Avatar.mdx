---
name: Avatar
route: /components/avatar
menu: Components
---

import { Playground } from 'docz';
import PropsTable from '../_docs/components/PropsTable';
import { Avatar } from './Avatar';
import Set from '../Set';

# Image

## Import

`import { Avatar } from 'fannypack'`

## Basic Usage

<Playground>
  <Avatar src="https://raw.githubusercontent.com/bigbellies/fannypack/master/src/_docs/bean.jpg" alt="Bean" fit="cover" height={100} width={100} />
</Playground>

## Avatar shapes

Avatars are available in two different shapes and can be changed using the `kind` prop.

<Playground>
  <Avatar src="https://raw.githubusercontent.com/bigbellies/fannypack/master/src/_docs/bean.jpg" kind="circle" alt="Bean" fit="cover" height={100} width={100} />
</Playground>

## Avatar sizes

Avatars are available in five different sizes or you can set your own using the `size` prop.

<Playground>
  <Set>
    <Avatar src="https://raw.githubusercontent.com/bigbellies/fannypack/master/src/_docs/bean.jpg" alt="Bean" kind="circle" fit="cover" size="xsmall" />
    <Avatar src="https://raw.githubusercontent.com/bigbellies/fannypack/master/src/_docs/bean.jpg" alt="Bean" kind="circle" fit="cover" size="small" />
    <Avatar src="https://raw.githubusercontent.com/bigbellies/fannypack/master/src/_docs/bean.jpg" alt="Bean" kind="circle" fit="cover" />
    <Avatar src="https://raw.githubusercontent.com/bigbellies/fannypack/master/src/_docs/bean.jpg" alt="Bean" kind="circle" fit="cover" size="medium" />
    <Avatar src="https://raw.githubusercontent.com/bigbellies/fannypack/master/src/_docs/bean.jpg" alt="Bean" kind="circle" fit="cover" size="large" />
    <Avatar src="https://raw.githubusercontent.com/bigbellies/fannypack/master/src/_docs/bean.jpg" alt="Bean" kind="circle" fit="cover" size={200} />
  </Set>
</Playground>

## Avatar using initials

Avatars can also be initials using the `initials` prop.

<Playground>
  <Set>
    <Avatar palette="primary" initials="MB" alt="Bean" kind="circle" size="xsmall" />
    <Avatar palette="success" initials="Mr Bean" alt="Bean" kind="circle" size="small" />
    <Avatar palette="primary" initials="MB" alt="Bean" />
    <Avatar palette="danger" initials="MB" alt="Bean" kind="circle" size="medium" />
    <Avatar palette="warning" initials="MB" alt="Bean" kind="circle" size="large" />
  </Set>
</Playground>

## Props

<PropsTable component={Avatar} />

## Theming

### Schema

```jsx
{
  Avatar: {
    // Styles to apply when the `fit` prop is present
    fit: {
      cover: css | Object,
      contain: css | Object
    },
    // Styles to apply for particular sizes
    sizes: {
      xsmall: css | Object,
      small: css | Object,
      medium: css | Object,
      large: css | Object
    }
  }
}
```
