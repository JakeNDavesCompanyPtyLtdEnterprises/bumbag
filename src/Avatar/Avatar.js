// @flow
import React, { Fragment } from 'react';
import classNames from 'classnames';
import _Avatar, { AvatarCircle, AvatarInitials } from './styled';

type AvatarType = 'circle' | 'square';

type Size = 'default' | 'xSmall' | 'small' | 'medium' | 'large';

type Props = {
  a11yLabel?: string,
  kind?: AvatarType,
  className?: string,
  fit?: 'cover' | 'contain',
  initials?: string,
  size?: Size,
  source?: string
};

const Avatar = ({ a11yLabel, className, initials, kind, size, source, ...props }: Props) => (
  <Fragment>
    {source && (
      <_Avatar
        alt={a11yLabel}
        className={classNames({
          [className || '']: Boolean(className)
        })}
        kind={kind}
        size={size}
        src={source}
        {...props}
      />
    )}
    {initials && (
      <AvatarCircle
        alt={a11yLabel}
        className={classNames({
          [className || '']: Boolean(className)
        })}
        kind={kind}
        size={size}
        {...props}
      >
        <AvatarInitials {...props}>
          {initials.split(' ').length === 2 ? initials.match(/\b\w/g) : initials.substring(0, 2)}
        </AvatarInitials>
      </AvatarCircle>
    )}
  </Fragment>
);

Avatar.defaultProps = {
  a11yLabel: undefined,
  kind: 'square',
  className: null,
  fit: null,
  initials: undefined,
  size: 'default',
  source: undefined
};

export default Avatar;
