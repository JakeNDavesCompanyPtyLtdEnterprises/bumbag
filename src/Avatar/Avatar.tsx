import * as React from 'react';
// @ts-ignore
import PropTypes from 'prop-types';
import classNames from 'classnames';
import _Avatar, { AvatarCircle } from './styled';
import { AvatarProps as ReakitAvatarProps } from 'reakit/ts';
import { Omit } from '../types';

export interface LocalAvatarProps {
  a11yLabel?: string;
  alt?: string;
  color?: string;
  className?: string;
  fit?: 'cover' | 'contain';
  fitPosition?: string;
  initials?: string;
  kind?: 'circle' | 'square';
  palette?: string;
  size?: number | 'default' | 'xsmall' | 'small' | 'medium' | 'large';
  src?: string;
}
export type AvatarProps = LocalAvatarProps & Omit<ReakitAvatarProps, 'size'>;

const defaultProps: Partial<LocalAvatarProps> = {
  a11yLabel: undefined,
  alt: undefined,
  kind: 'square',
  className: undefined,
  color: undefined,
  fit: undefined,
  fitPosition: undefined,
  initials: undefined,
  palette: undefined,
  size: undefined,
  src: undefined
};

export const Avatar: React.FunctionComponent<LocalAvatarProps> = ({
  a11yLabel,
  className,
  initials,
  size,
  src,
  ...props
}) => (
  <React.Fragment>
    {src && (
      <_Avatar
        alt={a11yLabel}
        className={classNames({
          [className || '']: Boolean(className)
        })}
        styledSize={size}
        src={src}
        {...props}
      />
    )}
    {initials && (
      <AvatarCircle
        className={classNames({
          [className || '']: Boolean(className)
        })}
        styledSize={size}
        {...props}
      >
        {initials.split(' ').length === 2 ? initials.match(/\b\w/g) : initials.substring(0, 2)}
      </AvatarCircle>
    )}
  </React.Fragment>
);

Avatar.propTypes = {
  a11yLabel: PropTypes.string,
  alt: PropTypes.string,
  kind: PropTypes.oneOf(['circle', 'square']),
  color: PropTypes.string,
  className: PropTypes.string,
  fit: PropTypes.oneOf(['cover', 'contain']),
  fitPosition: PropTypes.string,
  initials: PropTypes.string,
  palette: PropTypes.string,
  size: PropTypes.oneOfType([PropTypes.number, PropTypes.oneOf(['default', 'xsmall', 'small', 'medium', 'large'])]),
  src: PropTypes.string
};
Avatar.defaultProps = defaultProps;

export default Avatar as React.FunctionComponent<AvatarProps>;
