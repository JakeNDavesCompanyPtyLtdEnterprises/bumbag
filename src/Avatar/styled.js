// @flow
import styled, { css } from 'reakit/styled';
import { palette, theme } from 'styled-tools';
import _Avatar from 'reakit/Avatar';

const circleProperties = css`
  & {
    border-radius: 50%;
  }
`;

const fitProperties = props => {
  if (!props.fit) return;
  if (props.fit === 'contain') {
    return css`
      object-fit: contain;
      ${props =>
        props.fitPosition
          ? css`
              object-position: ${props.fitPosition};
            `
          : ''};

      ${theme('fannypack.Avatar.fit.base')};
      ${theme('fannypack.Avatar.fit.contain')};
    `;
  }
  if (props.fit === 'cover') {
    return css`
      object-fit: cover;
      ${props =>
        props.fitPosition
          ? css`
              object-position: ${props.fitPosition};
            `
          : ''};

      ${theme('fannypack.Avatar.fit.base')};
      ${theme('fannypack.Avatar.fit.cover')};
    `;
  }
};

const sizeProperties = {
  xsmall: css`
    & {
      width: 20px;
      height: 20px;
    }
    & {
      ${theme('fannypack.Avatar.sizes.xsmall')};
    }
  `,
  small: css`
    & {
      width: 40px;
      height: 40px;
    }
    & {
      ${theme('fannypack.Avatar.sizes.small')};
    }
  `,
  default: css`
    & {
      width: 60px;
      height: 60px;
    }
    & {
      ${theme('fannypack.Avatar.sizes.small')};
    }
  `,
  medium: css`
    & {
      width: 80px;
      height: 80px;
    }
    & {
      ${theme('fannypack.Avatar.sizes.medium')};
    }
  `,
  large: css`
    & {
      width: 160px;
      height: 160px;
    }
    & {
      ${theme('fannypack.Avatar.sizes.large')};
    }
  `
};

export const AvatarInitials = styled.div`
  background-color: ${props => palette(`${props.palette}Inverted`)(props)};
  ${props => props.kind === 'circle' && circleProperties};
  ${props => sizeProperties[props.size]};

  & {
    ${fitProperties};
  }

  ${theme('fannypack.AvatarInitials.base')};
`;

const Avatar = styled(_Avatar)`
  ${props => props.kind === 'circle' && circleProperties};
  ${props => sizeProperties[props.size]};

  & {
    ${fitProperties};
  }

  ${theme('fannypack.Avatar.base')};
`;

export default Avatar;
