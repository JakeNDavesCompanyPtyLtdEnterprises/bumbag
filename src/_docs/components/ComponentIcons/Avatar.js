import React from 'react';

const SvgComponent = props => (
  <svg width={117} height={52} fill="none" {...props}>
    <path
      d="M90.959 29.457c3.589 0 6.499-3.013 6.499-6.729S94.548 16 90.958 16c-3.588 0-6.498 3.012-6.498 6.728 0 3.716 2.91 6.729 6.499 6.729z"
      fill="#444bc9"
    />
    <path
      fillRule="evenodd"
      clipRule="evenodd"
      d="M103.956 45.058c0-7.432-5.819-13.457-12.997-13.457S77.96 37.626 77.96 45.058s25.995 7.432 25.995 0z"
      fill="#444bc9"
    />
    <path
      fillRule="evenodd"
      clipRule="evenodd"
      d="M90.904 51c13.807 0 25-11.193 25-25s-11.193-25-25-25c-1.535 0-3.037.138-4.496.403C74.747 3.521 65.904 13.728 65.904 26c0 13.807 11.193 25 25 25z"
      stroke="#444bc9"
    />
    <path
      d="M17.686 21.992h7.584v2.432h-4.144v2.224h3.84v2.016h-3.84V33h-3.44V21.992zm9.125 0h3.968c1.643 0 2.912.32 3.808.96.907.64 1.36 1.536 1.36 2.688 0 .725-.224 1.36-.672 1.904-.437.544-1.04.965-1.808 1.264-.757.288-1.6.432-2.528.432h-.672V33h-3.456V21.992zm4.144 5.024c.565 0 1.003-.117 1.312-.352.31-.235.464-.576.464-1.024 0-.448-.155-.784-.464-1.008-.299-.224-.736-.336-1.312-.336h-.688v2.688c.213.021.443.032.688.032z"
      fill="#3F1EAC"
    />
    <path
      fillRule="evenodd"
      clipRule="evenodd"
      d="M26 51.077c13.807 0 25-11.193 25-25s-11.193-25-25-25c-1.535 0-3.037.138-4.495.403C9.842 3.598 1 13.805 1 26.077c0 13.807 11.193 25 25 25z"
      stroke="#444bc9"
    />
  </svg>
);

export default SvgComponent;
