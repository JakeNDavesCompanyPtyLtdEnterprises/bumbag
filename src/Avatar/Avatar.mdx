---
name: Avatar
route: /components/avatar
menu: Components
---

import { Playground, PropsTable } from 'docz';
import Avatar from './index';
import Set from '../Set';

# Image

## Import

`import { Avatar } from 'fannypack'`

## Basic Usage

<Playground>
  <Avatar src="https://raw.githubusercontent.com/bigbellies/fannypack/master/src/_docs/bean.jpg" alt="Bean" fit="cover" height={100} width={100} />
</Playground>

## Avatar shapes

Avatars are available in two different shapes and can be changed using the `kind` prop.

<Playground>
  <Avatar src="https://raw.githubusercontent.com/bigbellies/fannypack/master/src/_docs/bean.jpg" kind="circle" alt="Bean" fit="cover" height={100} width={100} />
</Playground>

## Avatar sizes

Avatars are available in five different sizes and can be added with a `size` prop.

<Playground>
  <Set>
    <Avatar src="https://raw.githubusercontent.com/bigbellies/fannypack/master/src/_docs/bean.jpg" alt="Bean" kind="circle" fit="cover" size="xsmall" />
    <Avatar src="https://raw.githubusercontent.com/bigbellies/fannypack/master/src/_docs/bean.jpg" alt="Bean" kind="circle" fit="cover" size="small" />
    <Avatar src="https://raw.githubusercontent.com/bigbellies/fannypack/master/src/_docs/bean.jpg" alt="Bean" kind="circle" fit="cover" />
    <Avatar src="https://raw.githubusercontent.com/bigbellies/fannypack/master/src/_docs/bean.jpg" alt="Bean" kind="circle" fit="cover" size="medium" />
    <Avatar src="https://raw.githubusercontent.com/bigbellies/fannypack/master/src/_docs/bean.jpg" alt="Bean" kind="circle" fit="cover" size="large" />
  </Set>
</Playground>

## Avatar using initials

Avatars can also be initials using the `initials` prop.

<Playground>
  <Set>
    <Avatar palette="primary" initials="MB" alt="Bean" kind="circle" size="xsmall" />
    <Avatar palette="success" initials="Mr Bean" alt="Bean" kind="circle" size="small" />
    <Avatar palette="primary" initials="MB" alt="Bean" />
    <Avatar palette="danger" initials="MB" alt="Bean" kind="circle" size="medium" />
    <Avatar palette="warning" initials="MB" alt="Bean" kind="circle" size="large" />
  </Set>
</Playground>

## Props

<PropsTable of={Avatar} />

## Theming

### Schema

```jsx
{
  fit: {
    base: string | Object,
    cover: string | Object,
    contain: string | Object,
    size: string | Object
  }
}
```
