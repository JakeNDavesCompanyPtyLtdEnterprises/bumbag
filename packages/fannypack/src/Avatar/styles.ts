import { css, cssClass } from '../styled';
import { palette, theme } from '../utils';

export const Avatar = styleProps => cssClass`
  width: 60px;
  height: 60px;
  overflow: hidden;
  object-fit: cover;

  ${styleProps.kind === 'circle' &&
    css`
      border-radius: 50%;

      & {
        ${theme(`${styleProps.themeKey}.circle.root`)(styleProps)};
      }
    `}

  ${styleProps.initials &&
    css`
      background-color: ${palette(styleProps.palette)(styleProps)};
      color: ${palette(`${styleProps.palette}Inverted`)(styleProps)};
      display: flex;
      justify-content: center;
      align-items: center;
    `}

  ${getSizeAttributes(styleProps)};

  & {
    ${theme(`${styleProps.themeKey}.css.root`)(styleProps)};
  }
`;

function getSizeAttributes(styleProps) {
  const sizeAttributes = {
    small: css`
      width: 40px;
      height: 40px;

      & {
        ${theme(`${styleProps.themeKey}.css.sizes.small`)(styleProps)};
      }
    `,
    default: css`
      width: 60px;
      height: 60px;
      font-size: 24px;

      & {
        ${theme(`${styleProps.themeKey}.css.sizes.default`)(styleProps)};
      }
    `,
    medium: css`
      width: 80px;
      height: 80px;
      font-size: 36px;

      & {
        ${theme(`${styleProps.themeKey}.css.sizes.medium`)(styleProps)};
      }
    `,
    large: css`
      width: 100px;
      height: 100px;
      font-size: 48px;

      & {
        ${theme(`${styleProps.themeKey}.css.sizes.large`)(styleProps)};
      }
    `
  };
  return (
    sizeAttributes[styleProps.size] ||
    css`
      width: ${styleProps.size};
      height: ${styleProps.size};
    `
  );
}
