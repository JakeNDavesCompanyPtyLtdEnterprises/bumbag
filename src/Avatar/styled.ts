// @ts-ignore
import styled, { css } from 'reakit/styled';
import { palette, theme } from 'styled-tools';
// @ts-ignore
import _Avatar from 'reakit/Avatar';

import { AvatarProps } from './Avatar';
import { StyledProps } from '../types';

type Props = AvatarProps & StyledProps;

const circleProperties = css`
  & {
    border-radius: 50%;
  }
`;

const fitProperties = (props: Props) => {
  if (!props.fit) return;
  if (props.fit === 'contain') {
    return css`
      object-fit: contain;
      ${(props: Props) =>
        props.fitPosition
          ? css`
              object-position: ${props.fitPosition};
            `
          : ''};

      ${theme('fannypack.Avatar.fit.base')};
      ${theme('fannypack.Avatar.fit.contain')};
    `;
  }
  if (props.fit === 'cover') {
    return css`
      object-fit: cover;
      ${(props: Props) =>
        props.fitPosition
          ? css`
              object-position: ${props.fitPosition};
            `
          : ''};

      ${theme('fannypack.Avatar.fit.base')};
      ${theme('fannypack.Avatar.fit.cover')};
    `;
  }
};

const sizeProperties: { [key: string]: string } = {
  xsmall: css`
    & {
      font-size: 8px;
      width: 20px;
      height: 20px;
    }
    & {
      ${theme('fannypack.Avatar.sizes.xsmall')};
    }
  `,
  small: css`
    & {
      font-size: 16px;
      width: 40px;
      height: 40px;
    }
    & {
      ${theme('fannypack.Avatar.sizes.small')};
    }
  `,
  medium: css`
    & {
      font-size: 32px;
      width: 80px;
      height: 80px;
    }
    & {
      ${theme('fannypack.Avatar.sizes.medium')};
    }
  `,
  large: css`
    & {
      font-size: 48px;
      width: 100px;
      height: 100px;
    }
    & {
      ${theme('fannypack.Avatar.sizes.large')};
    }
  `
};

export const AvatarCircle = styled.div`
  align-items: center;
  background-color: ${(props: Props) => palette(props.palette || '')(props)};
  color: ${(props: Props) => (props.color ? props.color : palette(`${props.palette}Inverted`)(props))};
  display: flex;
  justify-content: center;
  font-size: 24px;
  height: ${(props: Props) => (typeof props.size === 'number' ? `${props.size}px` : '60px')};
  width: ${(props: Props) => (typeof props.size === 'number' ? `${props.size}px` : '60px')};

  ${(props: Props) => props.kind === 'circle' && circleProperties};
  ${(props: Props) => (typeof props.size === 'string' ? sizeProperties[props.size] : null)};

  ${theme('fannypack.Avatar.Circle.base')};
`;

const AvatarImage = styled(_Avatar)`
  font-size: 24px;
  height: ${(props: Props) => (typeof props.size === 'number' ? `${props.size}px` : '60px')};
  width: ${(props: Props) => (typeof props.size === 'number' ? `${props.size}px` : '60px')};
  ${(props: Props) => props.kind === 'circle' && circleProperties};
  ${(props: Props) => (typeof props.size === 'string' ? sizeProperties[props.size] : null)};

  & {
    ${fitProperties};
  }

  ${theme('fannypack.Avatar.Image.base')};
`;

export default AvatarImage;
