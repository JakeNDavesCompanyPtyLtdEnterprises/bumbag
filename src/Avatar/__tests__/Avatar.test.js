import React from 'react';
import render from '../../_utils/tests/render';
import Avatar from '../Avatar';
import 'jest-styled-components';

it('renders correctly for a default avatar', () => {
  const { container } = render(
    <Avatar source="https://raw.githubusercontent.com/bigbellies/fannypack/master/src/_docs/bean.jpg" alt="Mr bean" />
  );
  expect(container.firstChild).toMatchSnapshot();
});

it('renders correctly for an avatar with an initials prop set', () => {
  const { container } = render(<Avatar initials="SW" alt="SW" />);
  expect(container.firstChild).toMatchSnapshot();
});

it('renders correctly for an avatar with a size prop set', () => {
  const { container } = render(<Avatar initials="SW" size="large" alt="SW" />);
  expect(container.firstChild).toMatchSnapshot();
});

it('renders correctly for an avatar with a palette prop set', () => {
  const { container } = render(<Avatar initials="SW" palette="primary" alt="SW" />);
  expect(container.firstChild).toMatchSnapshot();
});

it('renders correctly for an avatar with a fit prop set', () => {
  const { container } = render(
    <Avatar
      source="https://raw.githubusercontent.com/bigbellies/fannypack/master/src/_docs/bean.jpg"
      fit="cover"
      alt="Mr bean"
    />
  );
  expect(container.firstChild).toMatchSnapshot();
});
