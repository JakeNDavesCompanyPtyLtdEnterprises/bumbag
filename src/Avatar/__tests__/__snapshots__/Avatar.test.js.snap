// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly for a default avatar 1`] = `
.c1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: inherit;
}

.c0 {
  box-sizing: border-box;
  font-family: system-ui,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Fira Sans', 'Droid Sans','Helvetica Neue',sans-serif;
  font-size: 16px;
  font-weight: 400;
  line-height: 1.5;
  margin: 0;
  padding: 0;
  overflow-x: hidden;
  -webkit-font-smoothing: antialiased;
  text-rendering: optimizeLegibility;
  color: #435a6f;
  fill: #435a6f;
}

.c0 *,
.c0 *::before,
.c0 *::after {
  box-sizing: inherit;
}

.c0 *:focus {
  outline: 2px solid #7467c0;
  outline-offset: 2px;
}

.c2 {
  font-size: 1.5em;
  width: 60px;
  height: 60px;
}

<div
  class="c0 c1"
>
  <img
    alt="Mr bean"
    class="c2 c1"
    kind="square"
    src="https://raw.githubusercontent.com/bigbellies/fannypack/master/src/_docs/bean.jpg"
    style="size: default;"
  />
</div>
`;

exports[`renders correctly for an avatar with a fit prop set 1`] = `
.c1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: inherit;
}

.c0 {
  box-sizing: border-box;
  font-family: system-ui,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Fira Sans', 'Droid Sans','Helvetica Neue',sans-serif;
  font-size: 16px;
  font-weight: 400;
  line-height: 1.5;
  margin: 0;
  padding: 0;
  overflow-x: hidden;
  -webkit-font-smoothing: antialiased;
  text-rendering: optimizeLegibility;
  color: #435a6f;
  fill: #435a6f;
}

.c0 *,
.c0 *::before,
.c0 *::after {
  box-sizing: inherit;
}

.c0 *:focus {
  outline: 2px solid #7467c0;
  outline-offset: 2px;
}

.c2 {
  font-size: 1.5em;
  width: 60px;
  height: 60px;
}

.c2 {
  object-fit: cover;
}

<div
  class="c0 c1"
>
  <img
    alt="Mr bean"
    class="c2 c1"
    kind="square"
    src="https://raw.githubusercontent.com/bigbellies/fannypack/master/src/_docs/bean.jpg"
    style="size: default;"
  />
</div>
`;

exports[`renders correctly for an avatar with a palette prop set 1`] = `
.c1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: inherit;
}

.c0 {
  box-sizing: border-box;
  font-family: system-ui,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Fira Sans', 'Droid Sans','Helvetica Neue',sans-serif;
  font-size: 16px;
  font-weight: 400;
  line-height: 1.5;
  margin: 0;
  padding: 0;
  overflow-x: hidden;
  -webkit-font-smoothing: antialiased;
  text-rendering: optimizeLegibility;
  color: #435a6f;
  fill: #435a6f;
}

.c0 *,
.c0 *::before,
.c0 *::after {
  box-sizing: inherit;
}

.c0 *:focus {
  outline: 2px solid #7467c0;
  outline-offset: 2px;
}

.c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #3926a5;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c2 {
  font-size: 1.5em;
  width: 60px;
  height: 60px;
}

.c3 {
  color: #fff;
}

<div
  class="c0 c1"
>
  <div
    alt="SW"
    class="c2"
    kind="square"
  >
    <div
      alt="SW"
      class="c3"
    >
      SW
    </div>
  </div>
</div>
`;

exports[`renders correctly for an avatar with a size prop set 1`] = `
.c1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: inherit;
}

.c0 {
  box-sizing: border-box;
  font-family: system-ui,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Fira Sans', 'Droid Sans','Helvetica Neue',sans-serif;
  font-size: 16px;
  font-weight: 400;
  line-height: 1.5;
  margin: 0;
  padding: 0;
  overflow-x: hidden;
  -webkit-font-smoothing: antialiased;
  text-rendering: optimizeLegibility;
  color: #435a6f;
  fill: #435a6f;
}

.c0 *,
.c0 *::before,
.c0 *::after {
  box-sizing: inherit;
}

.c0 *:focus {
  outline: 2px solid #7467c0;
  outline-offset: 2px;
}

.c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c2 {
  font-size: 3em;
  width: 100px;
  height: 100px;
}

<div
  class="c0 c1"
>
  <div
    alt="SW"
    class="c2"
    kind="square"
  >
    <div
      alt="SW"
      class=""
    >
      SW
    </div>
  </div>
</div>
`;

exports[`renders correctly for an avatar with an initials prop set 1`] = `
.c1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: inherit;
}

.c0 {
  box-sizing: border-box;
  font-family: system-ui,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Fira Sans', 'Droid Sans','Helvetica Neue',sans-serif;
  font-size: 16px;
  font-weight: 400;
  line-height: 1.5;
  margin: 0;
  padding: 0;
  overflow-x: hidden;
  -webkit-font-smoothing: antialiased;
  text-rendering: optimizeLegibility;
  color: #435a6f;
  fill: #435a6f;
}

.c0 *,
.c0 *::before,
.c0 *::after {
  box-sizing: inherit;
}

.c0 *:focus {
  outline: 2px solid #7467c0;
  outline-offset: 2px;
}

.c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c2 {
  font-size: 1.5em;
  width: 60px;
  height: 60px;
}

<div
  class="c0 c1"
>
  <div
    alt="SW"
    class="c2"
    kind="square"
  >
    <div
      alt="SW"
      class=""
    >
      SW
    </div>
  </div>
</div>
`;
